docs: add post-install steps to disable ssh password access (#783)
harvester/docs: docs/install/post-install.md

---
sidebar_position: 11
sidebar_label: Post-installation steps
title: "Post-Installation Steps"
keywords:
  - Harvester
  - Installation
description: Post-installation steps.
---

<head>
  <link rel="canonical" href="https://docs.harvesterhci.io/v1.6/install/post-install"/>
</head>

> These optional post-installation procedures describe how to enhance Harvester security and performance.
> 
> ## Disable SSH Password
> 
> By default during installation, SSH password authentication is enabled on the Harvester nodes. This allows administrator to access the nodes for installation diagnosis. 

Reviewer: ```suggestion
These optional post-installation procedures describe how to enhance Harvester security and performance.

You can enhance the security and performance of your Harvester cluster by performing the following procedures after installation is completed.

## Disable the SSH Password Authentication

By default, SSH password authentication is enabled on Harvester nodes during installation. This allows the administrator to access the nodes for post-installation diagnosis.
```

Author: Line 16 and 18 sounds a bit redundant. Can I just use the proposed line 18?

Reviewer: Sorry, line 16 is my wording and line 18 is yours. I apparently added instead of replaced. Feel free to use either.

> Once the installation completed successfully, it's recommended to disable SSH password authentication.
> 
> The following command uses `kubectl` to apply a [`CloudInit`](https://docs.harvesterhci.io/v1.6/advanced/cloudinitcrd/) configuration to disable SSH password authentication on all Harvester nodes:

Reviewer: ```suggestion
Once installation is completed, however, disabling SSH password authentication is recommended. You can run the following command, which uses `kubectl` to apply a [`CloudInit`](https://docs.harvesterhci.io/v1.6/advanced/cloudinitcrd/) configuration, to disable SSH password authentication on all Harvester nodes:
```

```sh
cat <<EOF | kubectl apply -f -
apiVersion: node.harvesterhci.io/v1beta1
kind: CloudInit
metadata:
  name: ssh-config
spec:
  matchSelector: 
    harvesterhci.io/managed: "true" # apply to all Harvester nodes
  filename: 99-ssh-config
  contents: |
    stages:
      network:
      - name: "disable password login"
        commands:
        - sed -i -E 's/^#?PasswordAuthentication .*/PasswordAuthentication no/' /etc/ssh/sshd_config
        - sed -i -E 's/^#?ChallengeResponseAuthentication .*/ChallengeResponseAuthentication no/' /etc/ssh/sshd_config
        - sed -i -E 's/^#?UsePAM .*/UsePAM no/' /etc/ssh/sshd_config
        - systemctl restart sshd
  paused: false
EOF
```

>    * The `matchSelector` field is used to select Harvester nodes with specific labels.
>    * All the affected nodes must be rebooted for the cloud-init configuration to take effect.
> 
> Once the configuration is applied, any attempts to access the Harvester nodes with the SSH password will be denied:

Reviewer: ```suggestion
:::note

- The `matchSelector` field is used to select Harvester nodes with specific labels.
- All the affected nodes must be rebooted for the `CloudInit` configuration to take effect.

:::

Once the configuration is applied, any attempts to access the Harvester nodes with the SSH password are denied.
```
